client/components/domains/registrant-extra-info/fr-form.jsx

```diff
@@ -40,82 +47,80 @@ class RegistrantExtraInfoForm extends React.PureComponent {
-		isProbablyOrganization: false,
-		values: {},
 		onSubmit: noop,
-		onStateChanged: noop,
 	}
 
-	componentWillMount() {
+	constructor( props ) {
+		super( props );
+
 		const defaults = {
-			registrantType: this.props.isProbablyOrganization
+			registrantType: this.props.contactDetails.organization
 				? 'organization' : 'individual',
-			countryOfBirth: this.props.countryCode || 'FR',
+			countryOfBirth: this.props.contactDetails.countryCode || 'FR',
 		};
 
-		const values = {
+		this.props.contactDetails.extra = {
 			...defaults,
-			...this.props.values,
+			...this.props.contactDetails.extra
 		};
+	}
 
-		// It's possible for the parent to pass in null and still have a form
-		// that the user is happy with, so we pass one state out
-		// immediatly to to make sure there's something valid in the parent
-		this.props.onStateChange( values );
+	componentDidUpdate( prevProps, prevState ) {
+		const dateOfBirth = this.compileDateOfBirth();
+
+		if ( dateOfBirth && ! isEqual( prevState, this.state ) ) {
+			this.setDateOfBirth( dateOfBirth );
+		}
 	}
 
 	handleDobChangeEvent = ( event ) => {
-		// TODO FIXME: Sanitize and validate individual fields and resulting
-		// date
+		this.setState( {
+			...this.state,
+			[ event.target.id ]: event.target.value,
+		} );
+	}
 
-		// setState() is not syncronous :(
-		const newState = this.newStateFromEvent( event );
-		const { dobYears, dobMonths, dobDays } = newState;
+	compileDateOfBirth() {
+		const { dobYears, dobMonths, dobDays } = this.state || {};
+		let dateOfBirth = false;
 
-		const dateOfBirth = ( dobYears && dobMonths && dobDays ) &&
-			[ dobYears, dobMonths, dobDays ].join( '-' );
+		if ( new Date().getFullYear() - toInteger( dobYears ) > currentPlausibleHumanLifespan ) {
+			// @todo: set error state
+			return false;
+		}
 
-		if ( dateOfBirth ) {
-			debug( 'Setting dateOfBirth to ' + dateOfBirth +
-				( moment( dateOfBirth, 'YYYY-MM-DD' ).isValid() ? '' : ' (invalid)' ) );
+		if ( dobYears && dobMonths && dobDays ) {
+			dateOfBirth = [ dobYears, dobMonths, dobDays ].join( '-' );
 		}
 
-		this.props.onStateChange( {
-			...newState,
-			...( dateOfBirth ? { dateOfBirth } : {} )
-		} );
-	}
+		if ( dateOfBirth && moment( dateOfBirth, 'YYYY-MM-DD' ).isValid() ) {
+			return dateOfBirth;
+		}
 
-	newStateFromEvent( event ) {
-		return {
-			...this.props.values,
-			[ event.target.id ]: event.target.value,
-		};
+		// @todo: set error state
+		return false;
 	}
 
-	handleChangeEvent = ( event ) => {
-		this.props.onStateChange( this.newStateFromEvent( event ) );
+	setDateOfBirth( dateOfBirth ) {
+		debug( 'Setting dateOfBirth to ' + dateOfBirth );
+		this.props.updateContactDetailsCache( { extra: { dateOfBirth } } );
 	}
 
-	// We need a deep comparison to check inside props.values
-	shouldComponentUpdate( nextProps ) {
-		return ! isEqual( this.props, nextProps );
+	handleChangeEvent = ( event ) => {
+		debug( 'Setting ' + event.target.id + ' to ' + event.target.value );
```

Comment:
Good candidate for ES6+ template strings:

```js
debug( `Setting ${ event.target.id } to ${ event.target.value }` );
```

Although note that `debug` takes as many args as needed, similarly to `console.log`. For instance:

```js
debug( 'setting id to value', event.target.id, event.target.value );
```

which would be useful to inspect values of non-primitive types.